### DISCUSSION TITLE
Getting a list of arrays

### DISCUSSION THREAD

**Discussion by paradajz** [TomWright/dasel#242] [Q&A] [answered]
Given the following JSON document:

```
{
  "array1": [
    {
      "a": "aaa",
      "b": "bbb",
      "c": "ccc"
    }
  ],
  "array2": [
    {
      "a": "aaa",
      "b": "bbb",
      "c": "ccc"
    }
  ]
}
```

How can I get a list of all arrays? In this case I am only interested in getting this as a result:

```
array1
array2
```

**Comment by TomWright** [comment, marked as answer]
You can use the index/property selector to return the keys of an object: https://dasel.tomwright.me/s/931c278a-a1e8-45f0-bc68-675daf6ebd46

```dasel -f in.json -m '.-'```

**Reply by paradajz** [reply]
Thank you!

**Comment by TomWright** [comment]
Hy @paradajz ,

With dasel v2 (still in development but you are welcome to start using it) you can do this with the following:
```bash
echo '{
  "array1": [
    {
      "a": "aaa",
      "b": "bbb",
      "c": "ccc"
    }
  ],
  "array2": [
    {
      "a": "aaa",
      "b": "bbb",
      "c": "ccc"
    }
  ]
}' | dasel -r json -w - 'all().filter(equal(type(),array)).key()'
array1
array2
```

V2 docs are here: https://daseldocs.tomwright.me/v/v2/
